ClientLib/src/main/java/com/o3dr/android/client/utils/data/tlog/TLogParser.java

```diff
@@ -0,0 +1,349 @@
+
+/**
+ * Parse TLog file into Events
+ */
+public class TLogParser {
+    private static final String LOG_TAG = TLogParser.class.getSimpleName();
+
+    private static final Parser parser = new Parser();
+    private static ExecutorService parseTlogExecutor;
+
+    public static class TLogIterator {
+        private File file;
+        private DataInputStream in = null;
+        private ExecutorService iteratorExecutor;
+        private final Handler handler;
+
+        /**
+         * Constructor to create a TLogIterator with new Handler.
+         *
+         * @param uri Location of the TLog files
+         */
+        public TLogIterator(URI uri) {
+            this(uri, new Handler());
+        }
+
+        /**
+         * Constructor to create a TLogIterator with a specified Handler.
+         *
+         * @param uri     Location of the TLog files
+         * @param handler Handler to post results to
+         */
+        public TLogIterator(URI uri, Handler handler) {
+            this.handler = handler;
+            file = new File(uri.toString());
+        }
+
+        /**
+         * Opens TLog file to begin iterating.
+         *
+         * @throws FileNotFoundException
+         */
+        public void start() throws FileNotFoundException {
+            iteratorExecutor = Executors.newSingleThreadExecutor();
+            in = new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
+        }
+
+        /**
+         * Closes TLog file
+         *
+         * @throws IOException
+         */
+        public void finish() throws IOException {
+            iteratorExecutor.shutdownNow();
+            in.close();
+        }
+
+        /**
+         * Retrieve next message from TLog file asynchronously.
+         * start must be called before this method.
+         *
+         * @param callback
+         */
+        public void nextAsync(final TLogIteratorCallback callback) {
+            iteratorExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        sendResult(callback, next(in));
+                    } catch (EOFException e) {
+                        sendResult(callback, null);
+                    } catch (IOException e) {
+                        sendFailed(callback, e);
+                    }
+                }
+            });
+        }
+
+        /**
+         * Retrieve next message with specified message filter from TLog file asynchronously.
+         * start must be called before this method.
+         *
+         * @param filter
+         * @param callback
+         */
+        public void nextAsync(final TLogIteratorFilter filter, final TLogIteratorCallback callback) {
+            iteratorExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Event event = null;
+                        while (in.available() > 0) {
```

Comment:
@chaviw The check for availability should be done by the `next(...)` method, and it should return `null` if no more data is available.